app/Http/Controllers/Backend/CategoryController.php

```diff
@@ -117,10 +119,28 @@ public function update(CategoryRequest $request, $id)
     /**
      * Remove the specified resource from storage.
      *
+     * @param \Illuminate\Http\Request $id id
+     *
      * @return \Illuminate\Http\Response
      */
-    public function destroy()
+    public function destroy($id)
     {
-        //
+        $cate = $this->category->find($id);
+        if (empty($cate)) {
+            return trans('messages.error_not_found');
+        }
+        
+        $promotions = $this->category->find($id)->promotions;
```

Comment:
[rank A] could we find other ways to make lower performance? you are using 2 queries in same table to check conditions only. I think it isn't good.